imports/plugins/included/ui-search/lib/containers/searchModalContainer.js

```diff
@@ -0,0 +1,266 @@
+import React, { Component } from "react";
+import { Meteor } from "meteor/meteor";
+import { Tracker } from "meteor/tracker";
+import { Roles } from "meteor/alanning:roles";
+import _ from "lodash";
+import { Reaction } from "/client/api";
+import { composeWithTracker } from "/lib/api/compose";
+import * as Collections from "/lib/collections";
+import SearchModal from "../components/searchModal";
+
+class SearchModalContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      collection: "products",
+      value: "",
+      tagResults: [],
+      productResults: [],
+      orderResults: [],
+      renderChild: true,
+      accountResults: [],
+      facets: []
+    };
+    this.handleClick = this.handleClick.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleChildUnmount = this.handleChildUnmount.bind(this);
+    this.handleAccountClick = this.handleAccountClick.bind(this);
+    this.handleOrderClick = this.handleOrderClick.bind(this);
+    this.handleTagClick = this.handleTagClick.bind(this);
+    this.handleToggle = this.handleToggle.bind(this);
+    this.dep = new Tracker.Dependency;
+  }
+
+  componentDidMount() {
+    Tracker.autorun(() => {
+      this.dep.depend();
+      const searchCollection = this.state.collection;
+      const searchQuery = this.state.value;
+      const facets = this.state.facets;
+      this.subscription = Meteor.subscribe("SearchResults", searchCollection, searchQuery, facets);
+
+      if (this.subscription.ready()) {
+        /*
+        * Product Search
+        */
+        if (searchCollection === "products") {
+          const productResults = Collections.ProductSearch.find().fetch();
+          this.setState({ productResults });
+
+          const productHashtags = getProductHashtags(productResults);
+          const tagSearchResults = Collections.Tags.find({
+            _id: { $in: productHashtags }
+          }).fetch();
+
+          this.setState({
+            tagResults: tagSearchResults,
+            accountResults: [],
+            orderResults: []
+          });
+        }
+
+        /*
+          * Account Search
+          */
+        if (searchCollection === "accounts") {
+          const accountResults = Collections.AccountSearch.find().fetch();
+          this.setState({
+            accountResults,
+            tagResults: [],
+            orderResults: [],
+            productResults: []
+          });
+        }
+
+        /*
+          * Order Search
+          */
+        if (searchCollection === "orders") {
+          const orderResults = Collections.OrderSearch.find(searchQuery, { limit: 99 }).fetch();
```

Comment:
This is why the Order search isn't working. You shouldn't be passing `searchQuery` to this collection as the client-side version of this collection is already search **results** only. And you are passing in a string value rather than a field/value pair anyway so that won't work. Was there a reason we aren't just doing a `find().fetch()` like the other ones?